### PR TITLE
added functionality and tests

### PR DESCRIPTION
**Pull request by franciscolopezsancho**
Allow passing metadata in testkits

References #501

### CODE REVIEW

File: samples/scala-eventsourced-counter/src/test/scala/com/example/actions/CounterJournalToTopicActionSpec.scala [hunk 1]
```diff
@@ -15,6 +18,13 @@ class CounterJournalToTopicActionSpec extends AnyWordSpec with Matchers {
       result.reply shouldBe Increased(1)
     }
 
+    "handle command Increase with Metadata" in {
+      val context = new TestKitActionContext(metadataEntries = Map("myKey" -> Left("myValue")))
```
Comment (franciscolopezsancho) [inline samples/scala-eventsourced-counter/src/test/scala/com/example/actions/CounterJournalToTopicActionSpec.scala:22]: I'm not sure I like a Left here but if not then I can't verify at compile time whether it's a String or a java.nio.ByteBuffer. 
Please, have a look at `com.akkaserveless.scalasdk.testkit.impl.TestKitActionContext`.
Another caviat is that `com.akkaserveless.javasdk.testkit.impl.TestKitActionContext` doesn't have Either so its constructor has a different signature.

File: samples/java-eventsourced-counter/src/test/java/com/example/actions/CounterJournalToTopicActionTest.java [hunk 2]
```diff
@@ -24,6 +28,14 @@ public void increaseTest() {
     assertEquals(1, result.getReply().getValue());
   }
 
+  @Test
+  public void increaseMetadataTest() {
+    TestKitActionContext context = new TestKitActionContext(new HashMap<String, String>(Map.of("myKey","myValue")));
```
Comment (johanandren) [inline samples/java-eventsourced-counter/src/test/java/com/example/actions/CounterJournalToTopicActionTest.java:33]: I think this is the wrong approach, Metadata is something that comes with each request, not something global across several calls.

I would have expected the user facing testkit API to add overloads to call methods specific metadata, like this:

```java
testKit.increase(ValueIncreased.newBuilder...build(), Metadata(stuff))
```

Comment (franciscolopezsancho) [reply samples/java-eventsourced-counter/src/test/java/com/example/actions/CounterJournalToTopicActionTest.java:33]: I see. Yep, now that you mention it, makes a lot of sense. I'll do that.

**Comment by franciscolopezsancho** [comment]
This approach is not adecuate.

**Pull request closed without merging**